From: Liam Howlett <liam.howlett@oracle.com>
Subject: maple_tree: fix mas_empty_area() wrapping
Date: Tue, 19 Jul 2022 17:01:30 +0000

Set the error code correctly when at the left most node and search has
been exhausted.  This was an overflow bug.

Link: https://lkml.kernel.org/r/20220719170118.3296447-1-Liam.Howlett@oracle.com
Signed-off-by: Liam R. Howlett <Liam.Howlett@oracle.com>
Reported-by: Sven Schnelle <svens@linux.ibm.com>
Tested-by: Sven Schnelle <svens@linux.ibm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/maple_tree.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/maple_tree.c~maple-tree-add-new-data-structure-fix-3
+++ a/lib/maple_tree.c
@@ -4994,7 +4994,7 @@ static inline bool mas_anode_descend(str
 		}
 next_slot:
 		min = pivot + 1;
-		if (mas->last < min) {
+		if (mas->last <= pivot) {
 			mas_set_err(mas, -EBUSY);
 			return true;
 		}
_
